cookbooks/aws-parallelcluster-platform/spec/unit/recipes/log_rotation_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'spec_helper'
+
+describe 'aws-parallelcluster-platform::log_rotation' do
+  for_all_oses do |platform, version|
+    context "on #{platform}#{version}" do
+      cached(:log_rotation_path) { "/etc/logrotate.d/parallelcluster_log_rotation" }
+      cached(:chef_run) do
+        runner(platform: platform, version: version).converge(described_recipe)
+      end
+      cached(:node) { chef_run.node }
+
+      context "in the head node when log_rotation enabled and dcv enabled and slurm" do
+        cached(:chef_run) do
+          runner = runner(platform: platform, version: version) do |node|
+            node.override['cluster']['node_type'] = "HeadNode"
+            node.override['cluster']['log_rotation_enabled'] = 'true'
+            node.override['cluster']['dcv_enabled'] = "head_node"
+            node.override['cluster']["directory_service"]["generate_ssh_keys_for_users"] = 'true'
+            node.override['cluster']["scheduler"] = 'slurm'
+            allow_any_instance_of(Object).to receive(:dcv_installed?).and_return(true)
+          end
+          runner.converge(described_recipe)
```

Comment:
As far as I know, this is converging 2 times: one time because if line 8 and the other one here.
You should converge in the most nested code snippet.

Reply:
You're right, I missed to remove the first converge (from a cut and paste).